src/components/select-menu/SelectMenuOption.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import classnames from 'classnames';
+
+import {SelectMenuOptionType} from './SelectMenu';
+import type {IconTypeType as SubjectIconTypeType} from '../subject-icons/SubjectIcon';
+import type {IconTypeType} from '../icons/Icon';
+
+import SubjectIcon from '../subject-icons/SubjectIcon';
+import Icon from '../icons/Icon';
+import Checkbox from '../form-elements/checkbox/Checkbox';
+import Text from '../text/Text';
+
+export type SelectOptionElementPropsType = {
+  option: SelectMenuOptionType;
+  isSelected?: boolean;
+  withIcon?: boolean;
+  multiSelect?: boolean;
```

Comment:
this shouldn't be an option of select option but should be defined on menuselect level and passed via context

Reply:
Ok i see why is that. In the near future we should migrate json based configuration to composition so the component is used more like:

```tsx
 <Select>
    <Select.Group label="Label">
      <Select.Option value="myvalue">Value</Select.Option>
    </Select.Group>
    <Select.Group label="Code">
      <Select.Option value="a">A/Select.Option>
      <Select.Option value="b">B</Select.Option>
      <Select.Option value="c">C</Select.Option>
    </Select.Group>
  </Select>
```